themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -1,41 +1,65 @@
 const _ = require('lodash');
 const dedent = require('dedent-tabs').default;
 
-const getTOC = (pages)=>{
+const getTOC = ()=>{
 
-	const recursiveAdd = (title, page, targetDepth, child, curDepth=0)=>{
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3, h4, h5, h6');
+	const headerDepth = ['H1', 'H2', 'H3', 'H4', 'H5', 'H6'];
+
+	const res = [];
+
+	const pageMap = [];
+
+	const walkPages = ()=>{
+		let current = 0;
+		let skip = 0;
+		let reset = 0;
+		const pages = iframeDocument.querySelectorAll('.page');
+		_.each(pages, (page)=>{
+			current++;
+			if(page.querySelectorAll('.skipCounting').length > 0) {
+				skip += 1;
+			} else if(page.querySelectorAll('.resetCounting').length > 0) {
```

Comment:
I don't think this accounts for the case where a page has both `.resetCounting` and `.skipCounting`. Might need to swap the order of the if statements here or remove the else? Not thinking straight what that logic needs to be right now.

You can see in a test document that the TOC will skip the reset and get out of sync.

Reply:
Yeah, I did not think about that scenario.